aptos-move/aptos-vm/src/move_vm_ext/vm.rs

```diff
@@ -87,17 +88,21 @@ impl MoveVmExt {
         }
 
         Ok(Self {
-            inner: MoveVM::new_with_config(aptos_natives_with_builder(&mut builder), VMConfig {
-                verifier: verifier_config,
-                max_binary_format_version,
-                paranoid_type_checks: crate::AptosVM::get_paranoid_checks(),
-                enable_invariant_violation_check_in_swap_loc,
-                type_size_limit,
-                max_value_nest_depth: Some(128),
-                type_max_cost,
-                type_base_cost,
-                type_byte_cost,
-            })?,
+            inner: WarmVmCache::get_warm_vm(
+                aptos_natives_with_builder(&mut builder),
```

Comment:
Instead of passing in the natives, it's better to pass in the required gas parameters, feature version, features, & timed features, so that we don't end up creating a bunch of unused native function closures in case of a cache hit.